core/src/peer_id.rs

```diff
@@ -53,11 +53,28 @@ impl PeerId {
     #[inline]
     pub fn from_bytes(data: Vec<u8>) -> Result<PeerId, Vec<u8>> {
         match multihash::Multihash::from_bytes(data) {
-            Ok(multihash) => Ok(PeerId { multihash }),
+            Ok(multihash) => {
+                if multihash.algorithm() == multihash::Hash::SHA2256 {
```

Comment:
`PeerId` can only ever be done with the SHA2-256-algo? I thought one of the points of multihashes prefixing was that it is "upgradable/flexible" on the hash used and doesn't need to be hard-coded. Why would we hard-code for this one algo only?

I'd see the point of not supporting old/broken hash-algorithms, but maybe we should check against those only...?

Reply:
That's a point I'm not sure about.
In practice all peer IDs always use SHA2-256. In theory I don't know.
One issue with having multihashes is that I can give you two hashes of the same data using two different algorithms, and they will not be identical.

Reply:
yeah, of course. And it is a rather annoying side-effect...

Let's merge it as is and put a ticket down, asking the question whether we should support other protocols...